temporian/implementation/numpy_cc/operators/tick_calendar.cc

```diff
@@ -0,0 +1,113 @@
+#include <pybind11/numpy.h>
+#include <pybind11/pybind11.h>
+
+#include <chrono>
+#include <cmath>
+#include <cstdint>
+#include <ctime>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "temporian/implementation/numpy_cc/operators/common.h"
+
+namespace {
+namespace py = pybind11;
+
+py::array_t<double> tick_calendar(
+    const double start_timestamp,                // min date
+    const double end_timestamp,                  // max date
+    const int min_second, const int max_second,  // second range
+    const int min_minute, const int max_minute,  // minute range
+    const int min_hour, const int max_hour,      // hours range
+    const int min_mday, const int max_mday,      // month days
+    const int min_month, const int max_month,    // month range
+    const int min_wday, const int max_wday       // weekdays
+) {
+  // Ticks list
+  std::vector<double> ticks;
+
+  // Date range
+  const long start_t = (long)std::floor(start_timestamp);
+  const long end_t = (long)std::floor(end_timestamp);
+
+  std::tm start_utc = *std::gmtime(&start_t);
+  int year = start_utc.tm_year;                           // from 1900
+  int month = std::max(start_utc.tm_mon + 1, min_month);  // zero-based tm_mon
+  int mday = std::max(start_utc.tm_mday, min_mday);       // 1-31
+  int hour = std::max(start_utc.tm_hour, min_hour);
+  int minute = std::max(start_utc.tm_min, min_minute);
+  int second = std::max(start_utc.tm_sec, min_second);
+
+  bool in_range = true;
+  while (in_range) {
+    while (month <= max_month && in_range) {
+      while (mday <= max_mday && in_range) {
+        while (hour <= max_hour && in_range) {
+          while (minute <= max_minute && in_range) {
+            while (second <= max_second && in_range) {
+              std::tm tm_struct = {};
+              tm_struct.tm_year = year;      // Since 1900
+              tm_struct.tm_mon = month - 1;  // zero-based
+              tm_struct.tm_mday = mday;
+              tm_struct.tm_hour = hour;
+              tm_struct.tm_min = minute;
+              tm_struct.tm_sec = second;
+
+              std::time_t time = std::mktime(&tm_struct);
```

Comment:
s/std::time_t/const std::time_t

Adding "const" to variables that are not expected to change can help readability :)